### ISSUE TITLE
install-EasyBuild-develop.sh cannot set upstream branch with new Git

### ISSUE THREAD

**Issue by zao** [easybuilders/easybuild-framework#2616]
```
zao@freja:~$ bash install-EasyBuild-develop.sh zao $EASYBUILD_PREFIX
=== Cloning zao/vsc-base ...
Cloning into 'vsc-base'...
remote: Enumerating objects: 1, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 5760 (delta 0), reused 0 (delta 0), pack-reused 5759
Receiving objects: 100% (5760/5760), 1.23 MiB | 1.12 MiB/s, done.
Resolving deltas: 100% (3412/3412), done.
=== Adding and fetching HPC-UGent GitHub repository @ hpcugent/vsc-base ...
From github.com:hpcugent/vsc-base
 * [new branch]      all_groups -> github_hpcugent/all_groups
 * [new branch]      master     -> github_hpcugent/master
fatal: the '--set-upstream' option is no longer supported. Please use '--track' or '--set-upstream-to' instead.
zao@freja:~$ git --version
git version 2.17.1
```

**Comment by zao** [comment]
Git man pages helpfully state:
```
       --set-upstream
           As this option had confusing syntax, it is no longer supported. Please use --track or --set-upstream-to instead.
```

**Comment by boegel** [comment]
Oh ffs... Any suggestions on how to fix this? Since which version of `git` are the alternatives supported?

**Comment by zao** [comment]
All of `--track`, `--no-track`, `--set-upstream-to` are defined since at least [2.4.0](https://git-scm.com/docs/git-branch/2.4.0) (2015-04-30).

`--set-upstream` was deprecated back in 2012 with the [1.8.0 release notes](https://github.com/git/git/blob/53f9a3e157dbbc901a02ac2c73346d375e24978c/Documentation/RelNotes/1.8.0.txt#L20-L22) which also mention that `--set-upstream-to` was added and intended to replace lost functionality.

**Comment by boegel** [comment]
@zao Should be fixed with #2651, please test?

**Comment by zao** [comment]
The new option seems to have the same semantics as the old option, looks good.

I was tricked for a bit where the message about `git pull origin` in the repos at the end of the installation procedure doesn't seem overly helpful, but it seems like that held for older Git versions as well.

**Comment by boegel** [comment]
fixed in #2651
